iradix.go

```diff
@@ -86,6 +86,20 @@ func (t *Tree) Txn() *Txn {
 	return txn
 }
 
+// Clone makes an independent copy of the transaction. The new transaction
+// does not track any nodes and has TrackMutate turned off.
```

Comment:
My main concern with this is that a transaction is not thread safe. All of the assignments below are not done in a concurrency-safe manner to you could end up with an inconsistent set of values in the new txn -- or worse, on the existing transaction, the write to t.writable could cause a data race.

Reply:
Is this principally different from the assignment to `t.writable` in `writeNode`?

Reply:
Good question. I assumed you'd want to Clone so you can handle transactions in different threads. But I guess this package isn't actually thread safe to begin with?

@banks any insight here?

Reply:
I'm not treating it as thread-safe in the first place. I assume that a single thread owns a transaction. `Clone` must be called by the transaction's owning thread; the clone can be owned by a different thread afterwards.

Reply:
I don't think it can be though, because you're referencing the same Node for root and snap. If you pass the transaction to another thread you have two threads potentially modifying those values.

Reply:
Nope, the new transaction is read-only and just contains a pointer to an immutable node.

Reply:
I mean transfer it irreversibly (without touching it again in the original thread). Only one thread touches any particular transaction. So there will never be a race to modify the `root` and `snap` fields of any given `Txn`. And nodes themselves are immutable, so safe to share.

Reply:
```suggestion
// does not track any nodes and has TrackMutate turned off. The cloned transaction will contain any uncommitted writes in the original transaction but further mutations to either will be independent and result in different radix trees on Commit. A cloned transaction may be passed to another goroutine and mutated there independently however each transaction may only be mutated in a single thread.
```

This is a bit clumsy but I think that's useful clarification for users. What do you think?